pkg/util/subresource.go

```diff
@@ -61,6 +62,11 @@ func GetAgentDetailsValue(h handler, key string) (string, error) {
 // Returns errors if unable to perform type conversion.
 // Returns an empty string if the value does not exist, or if there is an error.
 func GetSubResourcePropertyValue(h handler, subRes, key string) (string, error) {
+	// check for a nil value, or a pointer to a nil value
+	if h == nil || reflect.ValueOf(h).Kind() == reflect.Ptr && reflect.ValueOf(h).IsNil() {
```

Comment:
Checking for nil, or a nil pointer to avoid panics

Reply:
maybe this check is worth moving to util function?

Reply:
I like that idea